### PR TITLE
PR1 | Generated Trigger class files

### PR DESCRIPTION
**Pull request by MounikaBattu17**
**Why should this Pull Request be merged?**
- This PR generates triggering folder files (`arm_start_trigger.py`, `handshake_trigger.py`, `pause_trigger.py`, `reference_trigger.py`, `start_trigger.py`)

**What does this Pull Request accomplish?**
- This implements 
[Task 2316158](https://dev.azure.com/ni/DevCentral/_workitems/edit/2316158): generate the trigger related classes

### CODE REVIEW

**Comment by MounikaBattu17** [comment]
The differences in the generated trigger files given below:
- `arm_start_trigger.py`: `timescale` is changed to `time_timescale`
- `handshake_trigger.py`: No changes 
- `pause_trigger.py`:  No changes 
- `reference_trigger.py`:  No changes 
- `start_trigger.py`: `timescale` is changed to `time_timescale`

**Review by zhindes — CHANGES_REQUESTED** [review]

File: generated/nidaqmx/_task_modules/triggering/start_trigger.py [hunk 1]
```diff
@@ -2252,7 +2252,7 @@ def term(self):
         return val.value.decode('ascii')
 
     @property
-    def time_timescale(self):
+    def timescale(self):
```
[the remark this answers is not in the document]
Comment (MounikaBattu17) [reply generated/nidaqmx/_task_modules/triggering/start_trigger.py]: Updated naming change from timescale to time_timescale and removed the timescale in deprecation logic.

File: src/codegen/utilities/attribute_helpers.py [hunk 2]
```diff
@@ -112,6 +141,13 @@ def get_attributes(metadata, class_name):
                 and attribute_data["python_class_name"] == class_name
                 and not attribute_data["name"] in EXCLUDED_ATTRIBUTES
             ):
+                # Strip class name in attribute name from the description.
+                attribute_data["python_description"] = _strip_attr_name_in_description(
+                    attribute_data["python_description"]
+                )
+
+                # Strip class name in the attribute name.
+                attribute_data["name"] = _strip_name(attribute_data["name"], class_name)
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply src/codegen/utilities/attribute_helpers.py:150]: Also:
- The `Attribute` class does some name fixups

```
class Attribute:
    """Structure for storing attribute metadata from scrapigen."""
    ATTRIBUTE_CHANGE_SET = {
        "ai_custom_scale_name": "ai_custom_scale",
        "ao_custom_scale_name": "ao_custom_scale",
        "ci_custom_scale_name": "ci_custom_scale",
        "ci_dup_count_prevent": "ci_dup_count_prevention",
        "chan_descr": "description",
        "chan_sync_unlock_behavior": "sync_unlock_behavior",
        "chan_is_global": "is_global",
        "physical_chan_name": "physical_channel",
    }
```

- On further thought, those name fixups also belong in scrapigen

File: README.rst [hunk 3]
```diff
@@ -32,9 +32,13 @@ system.
...
 
-Running **nidaqmx** requires NI-DAQmx or NI-DAQmx Runtime. Visit the
+Running **nidaqmx** requires NI-DAQmx to be installed. Visit
 `ni.com/downloads <http://www.ni.com/downloads/>`_ to download the latest
-version of NI-DAQmx.
+version of NI-DAQmx. None of the recommended **Additional items** are required
+for **nidaqmx** to function, and they can be removed to minimize installation
+size. It is recommended you continue to install the **NI Certificates** package
+to allow your Operating System to trust NI built binaries, improving your
+software and hardware installation experience.
```
Comment (zhindes) [inline README.rst]: These changes are already in master... the diffs here are really strange. can you rebase your commits on top of the current master?

File: generated/nidaqmx/_task_modules/channels/ci_channel.py [hunk 4]
```diff
@@ -12842,3 +12842,168 @@ def ci_velocity_meas_time(self):
             self._handle, self._name)
         check_for_error(error_code)
 
+    @property
```
Comment (zhindes) [inline generated/nidaqmx/_task_modules/channels/ci_channel.py]: These commits are also weird... the deprecations aren't valid. `ci_count_edges_count_reset_reset_cnt` still exists.

**Review by zhindes — CHANGES_REQUESTED** [review]
something weird is going on with the commits, rebase perhaps?

File: generated/nidaqmx/_task_modules/channels/ci_channel.py (hunk 4 above)
Comment (zhindes) [reply generated/nidaqmx/_task_modules/channels/ci_channel.py]: Latest commits fix it, woo!

File: README.rst (hunk 3 above)
Comment (zhindes) [reply README.rst]: Latest commits fix it, woo!

**Review by zhindes — APPROVED** [review]

File: src/codegen/utilities/attribute_helpers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MounikaBattu17) [reply src/codegen/utilities/attribute_helpers.py:191]: As suggested incorrect `python_description` will be fixed later. Created a [Technical Debt 2347643](https://dev.azure.com/ni/DevCentral/_workitems/edit/2347643): Fix incorrect "python_description" in attributes metadata to track this issue.

**Review by bkeryan — APPROVED** [review]

File: src/codegen/utilities/attribute_helpers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MounikaBattu17) [reply src/codegen/utilities/attribute_helpers.py:196]: Resolving this conversion.

**Comment by MounikaBattu17** [comment]
@zhindes This PR has got required approvals. Kindly merge.

**Comment by zhindes** [comment]
@MounikaBattu17 - merge conflict on ` src/codegen/metadata/script_info.py`

**Comment by MounikaBattu17** [comment]
> @MounikaBattu17 - merge conflict on ` src/codegen/metadata/script_info.py`

@zhindes  Resolved the merge conflicts. Please merge.

**Pull request merged**